frontend/packages/knative-plugin/src/topology/knative-topology-utils.ts

```diff
@@ -60,6 +60,14 @@ type RevK8sResourceKind = K8sResourceKind & {
 export const getEventSourceStatus = ({ FLAGS }: RootState): boolean =>
   FLAGS.get(FLAG_KNATIVE_EVENTING);
 
+/**
+ * get knative service routes url based on the revision's traffic
+ */
+export const getKnativeServiceRoutesURL = (ksvc: K8sResourceKind): string => {
+  const maximumTraffic: TrafficData = _.maxBy(ksvc.status.traffic as TrafficData[], 'percent');
```

Comment:
at times `status` may not be there so would safe to have `ksvc.status?.traffic`

Reply:
fixed, added additional checks on the top of the function.